lib/ar2dto/dto.rb

```diff
@@ -18,9 +18,16 @@ def self.inherited(base)
     end
 
     def initialize(attributes = {})
-      singleton_class.instance_eval { attr_reader(*attributes.keys) }
+      regular_attributes = attributes.reject { |key, _| key.to_s.end_with?("?") }
+      singleton_class.instance_eval { attr_reader(*regular_attributes.keys) }
+      regular_attributes.each { |key, value| instance_variable_set("@#{key}", value) }
 
-      attributes.each { |key, value| instance_variable_set("@#{key}", value) }
+      boolean_attributes = attributes.select { |key, _| key.to_s.end_with?("?") }
+      boolean_attributes.each do |key, value|
+        define_singleton_method key.to_s do
```

Comment:
It works, but not sure if this solution has any downsides.

If it's Ok, then I will proceed to refactor this to make it look better if possible 😅

Reply:
I think this approach is fine, but maybe we don't need to differentiate between boolean and not booleans and we can set all the values as methods instead of instance variables?

Reply:
Agree with that proposal 👍 
I didn't know whether keeping the attributes as instance variables could have any benefits over defining everything as methods.

Reply:
hmm now that you mention, I'm not sure how this can affect the `.as_json` method on the DTO, and it is being used on the `eq` method. Might be worth checking